index.js

```diff
@@ -39,36 +53,33 @@ class RequestError extends Error {
  *
  * @param  {Function} config.logger         logging function, defaulted to noop ( () => {} )
  * @param  {Boolean}  config.json           respond with JSON or string, default true
- * @param  {Boolean}  config.catchAll       Catch all errors or call next middleware?
- * @param  {Boolean}  config.endRequest     End the request?
+ * @param  {Boolean}  config.catchAll       Catch all errors (and end the request) or call next middleware?
  * @param  {Mixed}    config.defaultFail    Default value to respond with when the server throws a 500 and no
  *                                          body is specified
  * @return {Function}                       Middleware function
  */
+const catchMiddleware = function (config) {
 
-const catchMiddleware = function(config) {
-    return function(err, req, res, next) {
-        if ('boolean' !== typeof config.json) {
-            debug('config.json not set, defaulting to true');
-            config.json = true;
-        }
-        if ('boolean' !== typeof config.catchAll) {
-            debug('config.catchAll is not set, defaulting to false');
-            config.catchAll = false;
-        }
-        if ('boolean' !== typeof config.endRequest) {
-            debug('config.endRequest is not set, defaulting to true');
-            config.endRequest = true;
-        }
-        if (!config.defaultFail) {
-            debug(`config.defaultFail is not set, defaulting it to ${config.json ? '{}' : '\'\''}`);
-            config.defaultFail = config.json ? {} : '';
-        }
+    if ('boolean' !== typeof config.json) {
+        debug('config.json not set, defaulting to true');
+        config.json = true;
+    }
+    if ('boolean' !== typeof config.catchAll) {
+        debug('config.catchAll is not set, defaulting to false');
+        config.catchAll = false;
+    }
+    if (!config.defaultFail) {
+        debug(`config.defaultFail is not set, defaulting it to ${config.json ? '{}' : '\'\''}`);
+        config.defaultFail = config.json ? {} : '';
+    }
+
+    const fn = config.json ? 'json' : 'send';
 
-        const fn = config.json ? 'json' : 'send';
+    if (config.logger && 'function' !== typeof config.logger) {
```

Comment:
I don't like how you added the brackets around the single-line `if` here, but in line 84 they are not there, that's inconsistent. Can you remove them here?